src/Web.cpp

```diff
@@ -1732,33 +1742,38 @@ static void handleGetRFIDRequest(AsyncWebServerRequest *request) {
 	// construct chunked repsonse
 	nvsIndex = 0;
 	AsyncWebServerResponse *response = request->beginChunkedResponse("application/json",
-		[](uint8_t *buffer, size_t maxLen, size_t index) {
+		[nvsKeys = std::move(nvsKeys), withDetails](uint8_t *buffer, size_t maxLen, size_t index) {
 			maxLen = maxLen >> 1; // some sort of bug with actual size available, reduce the len
 			size_t len = 0;
-			static String json;
+			String json;
 
 			if (nvsIndex == 0) {
 				// start, write first tag
-				json = tagIdToJsonStr(nvsKeys[nvsIndex].c_str());
-				len += sprintf(((char *) buffer), "[%s", json.c_str());
+				json = tagIdToJsonStr(nvsKeys[nvsIndex].c_str(), withDetails);
+				if (json.length() > maxLen) {
```

Comment:
You check that, but 4 lines later you're writing json.length() + 1 bytes, so you should also check for json.length() + 1 available space